features/promotion/reapplying_promotion_on_cart_change.feature

```diff
@@ -0,0 +1,60 @@
+@promotion
+Feature: Reapplying promotion on cart change
+  In order to receive proper discount for my order
+  As a Customer
+  I want to have proper discount applied after every operation on my cart
+
+  Background:
+    Given the store is operating on a single "France" channel
+    And the store has a product "PHP T-Shirt" priced at "€100.00"
+    And there is a promotion "Holiday promotion"
+    And I am logged in customer
+
+  @todo
+  Scenario: Not applying promotion on shipping on empty cart after product removal
+    Given the store has "DHL" shipping method with "€10.00" fee
+    And the promotion gives "100%" percentage discount on shipping to every order
+    And I have product "PHP T-Shirt" in the cart
+    When I proceed selecting "DHL" shipping method
+    And I remove product "PHP T-Shirt" from the cart
+    Then my cart total should be "€0.00"
+    And my cart shipping fee should be "€0.00"
+    And my cart promotions should be "€0.00"
```

Comment:
And my discount should be
And discount on my cart should be

I think I prefer first one.